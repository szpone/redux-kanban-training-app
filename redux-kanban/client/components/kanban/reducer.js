

export function left(id) {
    return {
        type: 'LEFT',
        id,
    }
}

export function right(id) {
    return {
        type: 'RIGHT',
        id,
    }
}

export function add(name) {
    return {
        type: 'ADD',
        name,
    }
}


const initialState = {
    "new": [{
        "name": "kotek",
        "id": 1
    }],
    "done": [{
        "name": "piesek",
        "id": 2
    }],
    "nextId": 3,
};

export default function reducer(state = initialState, action) {
    state = {
        new: state.new.slice(),
        done: state.done.slice(),
        nextId: state.nextId,
    };
    switch(action.type) {
        case 'RIGHT': {
            for (let i in state.new) {
                let item = state.new[i];
                if (item.id === action.id) {
                    state.new.splice(i, 1);
                    state.done.push(item);
                    break;
                }
            }
            return state;
        }

        case 'LEFT': {
            for (let i in state.done) {
                let item = state.done[i];
                if (item.id === action.id) {
                    state.done.splice(i, 1);
                    state.new.push(item);
                    break;
                }
            }
            return state;
        }

        case 'ADD': {
            state.new.push({"name": action.name, "id": state.nextId});
            state.nextId++;
            return state;
        }
        default:
            return state;
    }
}


