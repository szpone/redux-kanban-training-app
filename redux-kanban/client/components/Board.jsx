import React from 'react';
import { connect } from 'react-redux';
import { left, right } from './kanban/reducer.js';
import TaskForm from './TaskForm.jsx';

export class BoardComponent extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { onLeft, onRight, board } = this.props;
        return (
            <div>
                <Column name="Nowe" items={board.new} button={onRight} />
                <Column name="Zrobione" items={board.done} button={onLeft} />
                <TaskForm/>
            </div>
        )
    }
}

export const Column = (props) => {
    const { name, items, button } = props;
    return (
        <div>
            <h1>{name}</h1>
            <ul>
                {items.map((item, i) => <li key={i}>{item.name}
                <button onClick={()=>button(item.id)}>Done</button></li>)}
            </ul>
        </div>
    )

};


function mapStateToProps(state) {
    return {
        board: state,
    }
}

function mapDispatchToProps(dispatch) {
    return {
        onLeft(id){
            dispatch(left(id))
        },
        onRight(id) {
            dispatch(right(id))
        }
    }
}

const Board = connect(
    mapStateToProps,
    mapDispatchToProps
)(BoardComponent);

export default Board;
