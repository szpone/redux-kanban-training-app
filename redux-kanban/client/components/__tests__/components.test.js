import React from 'react';
import ReactDOM from 'react-dom';
import ConnectedTaskForm, { TaskForm } from '../TaskForm.jsx';
import ConnectedBoard, { Board } from '../Board.jsx';
import { Column, BoardComponent } from '../Board.jsx';
import { connect } from 'react-redux'
import { Provider } from 'react-redux';
import {createStore} from "redux";
import reducer from "../kanban/reducer";


let store = createStore(
    reducer,
);

test('render a form', () => {
   const wrapper = shallow(
       <Provider store={store}><ConnectedTaskForm/></Provider>
   );
   expect(wrapper).toMatchSnapshot();
});


describe('<Column />', () => {
   it('allows to set props', () => {
       const wrapper = mount(<Column name="Nowe" />);
       expect(wrapper.props().name).to.equal("Nowe");
   })
});
