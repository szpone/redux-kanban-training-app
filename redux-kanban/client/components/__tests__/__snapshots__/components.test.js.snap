// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a form 1`] = `<Connect(TaskFormComponent) />`;
